tools/convert/main.cpp

```diff
@@ -270,6 +275,18 @@ int main(int argc, const char** argv)
             auto dx12_json_flags = output_format == JsonFormat::JSON ? gfxrecon::util::kToString_Formatted
                                                                      : gfxrecon::util::kToString_Unformatted;
             dx12_ascii_consumer.Initialize(out_file_handle, dx12_json_flags);
+
+#ifdef GFXRECON_AGS_SUPPORT
+            gfxrecon::decode::AgsAsciiConsumer ags_ascii_consumer;
+            ags_ascii_consumer.Initialize(out_file_handle, dx12_json_flags);
+
+            gfxrecon::decode::AgsDecoder ags_decoder;
+
+            ags_decoder.AddConsumer(reinterpret_cast<gfxrecon::decode::AgsConsumerBase*>(&ags_ascii_consumer));
```

Comment:
This cast should not be needed, and definitely not a `reinterpret_cast` unless I'm missing something here. Casting a class to its public base class makes no sense. See: `class AgsAsciiConsumer : public AgsConsumerBase`.

This should be fine:
` ags_decoder.AddConsumer(&ags_ascii_consumer);`